src/mpid/ch4/shm/posix/posix_am.h

```diff
@@ -306,7 +306,8 @@ MPL_STATIC_INLINE_PREFIX int MPIDI_POSIX_do_am_isend(int grank,
 
     send_size = MPIDIG_am_send_async_get_data_sz_left(sreq);
     offset = MPIDIG_am_send_async_get_offset(sreq);
-    MPI_Aint *bytes_sent_out = MPIDIG_am_send_async_get_data_sz_left(sreq) ? &send_size : NULL;
+    MPI_Aint *bytes_sent_out = send_size ? &send_size : NULL;
```

Comment:
this patch is good, but is anything every checking `bytes_sent_out`?

Reply:
It's a pointer to `send_size`, which is updated by `MPIDI_POSIX_eager_send`. But it's a good point. I don't see why we need worry about passing `NULL` vs passing `&send_size` all the time. Let me change that.

Reply:
Oh, now I see it. Yes, probably can be improved 😄.